Library/Homebrew/utils/service.rb

```diff
@@ -0,0 +1,50 @@
+# typed: true
+# frozen_string_literal: true
+
+module Utils
+  # Helpers for `brew services` related code.
+  module Service
+    # Check if a service is running for a specified formula.
+    sig { params(formula: Formula).returns(T::Boolean) }
+    def self.running?(formula)
+      if launchctl?
+        quiet_system(launchctl, "list", formula.plist_name)
+      elsif systemctl?
+        quiet_system(systemctl, "is-active", "--quiet", formula.service_name)
```

Comment:
I found this online when looking up how to test for active systemd services but as I'm on MacOS. I'd appreciate it if someone with a Linux box could confirm that this work correctly.

An easy way to check would be to check out this branch and run `brew info <formula>` on two formulas with service blocks. One should be installed with a service running and the other shouldn't be running whether installed or not. The caveats message should change; the running one will require a restart while the other will not.

Reply:
When it's running...
```
To restart redis after an upgrade:
  brew services restart redis
```
...and when it's not.
```
To start redis now and restart at login:
  brew services start redis
```